src/erldns_resolver.erl

```diff
@@ -458,28 +464,42 @@ best_match(Qname, Labels, Zone, []) ->
   end;
 best_match(_Qname, _Labels, _Zone, WildcardMatches) -> WildcardMatches.
 
-
-
 %% Function for executing custom lookups by registered handlers.
--spec custom_lookup(dns:dname(), dns:type(), [dns:rr()]) -> fun(({module(), [dns:type()]}) -> [dns:rr()]).
-custom_lookup(Qname, Qtype, Records) ->
-  fun({Module, Types}) ->
-      case lists:member(Qtype, Types) of
-        true -> Module:handle(Qname, Qtype, Records);
-        false ->
-          case Qtype =:= ?DNS_TYPE_ANY of
-            true -> Module:handle(Qname, Qtype, Records);
-            false -> []
-          end
-      end
+-spec custom_lookup(dns:dname(), dns:type(), [dns:rr()], dns:message()) -> fun(({module(), [dns:type()], integer()}) -> [dns:rr()]).
+custom_lookup(Qname, Qtype, Records, Message) -> 
+  fun({Module, Types, Version}) ->
+	  case Version of
+		  1 ->
+			  case lists:member(Qtype, Types) of
+				true -> Module:handle(Qname, Qtype, Records);
+				false ->
+				  case Qtype =:= ?DNS_TYPE_ANY of
+					true -> Module:handle(Qname, Qtype, Records);
+					false -> []
+				  end
+			  end;
+		  2 -> 
+			  case lists:member(Qtype, Types) of
+				true -> Module:handle(Qname, Qtype, Records, Message);
+				false ->
+				  case Qtype =:= ?DNS_TYPE_ANY of
+					true -> Module:handle(Qname, Qtype, Records, Message);
+					false -> []
+				  end
+			  end;
+		  _ -> []
```

Comment:
Is this condition ever expected to be reached? If not, then can this case be removed so the call will result in an error rather than returning silently?

Reply:
left default case with the idea that if someone registers handler with the wrong/unhandled version (=/= [1,2]), the custom lookup will continue to function - you are right, it's probably better to yield an error so they can address this - amending now.

register_handler([?DNS_TYPE_A], ?MODULE, X),